lib/prettify/index.ts

```diff
@@ -1,5 +1,8 @@
 export type Prettify<Type> = Type extends Function
   ? Type
-  : {
-      [Key in keyof Type]: Type[Key];
-    } & {};
+  : Extract<
```

Comment:
Great optimisation on the assignability, thank you for giving a great example in Playground